File: Level 2/Level_2_멀쩡한 사각형.cpp
/* 프로그래머스 - Level 2 멀쩡한 사각형 */

using namespace std;

int gcd(long long a, long long b) {
	return (a % b == 0) ? b : gcd(b, a % b);
}

long long solution(int w, int h) {
	long long lw = w, lh = h;
	long long answer = lw * lh - (lw + lh - gcd(lw, lh));

	return answer;
}

int main() {
	solution(8, 12);

	return 0;
}

File: Level 2/Level_2_멀쩡한 사각형.kt
/* 프로그래머스 - Level 2 멀쩡한 사각형*/

package level2

class Solution {
    private fun gcd(op1: Long, op2: Long): Long = if (op1 % op2 == 0L) op2 else gcd(op2, op1 % op2)

    fun solution(w: Int, h: Int): Long {
        val width: Long = w.toLong()
        val height: Long = h.toLong()

        return width * height - (width + height - gcd(width, height))
    }
}